lib/modules/datasource/docker/index.ts

```diff
@@ -812,54 +812,144 @@ export class DockerDatasource extends Datasource {
   @cache({
     namespace: 'datasource-docker-digest',
     key: (
-      { registryUrl, packageName }: GetReleasesConfig,
+      { registryUrl, packageName, currentDigest }: DigestConfig,
       newValue?: string
     ) => {
       const newTag = newValue ?? 'latest';
       const { registryHost, dockerRepository } = getRegistryRepository(
         packageName,
         registryUrl!
       );
-      return `${registryHost}:${dockerRepository}:${newTag}`;
+      const digest = currentDigest ? '@' + currentDigest : '';
+      return `${registryHost}:${dockerRepository}:${newTag}${digest}`;
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
```

Comment:
why this change?

Reply:
To get rid of an eslint warning: `Invalid type "string | undefined" of template literal expression  @typescript-eslint/restrict-template-expressions`

This is shown because `newValue` can be null and was implicitly casted to a string.